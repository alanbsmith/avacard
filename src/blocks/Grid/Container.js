import styled from "styled-components";
import { applyStyleModifiers } from 'styled-components-modifiers';

import flex from "../../modifiers/flex";
import { gridScale } from "../../utils";

const modifiers = {
  ...flex,
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${gridScale(1)};
  ${applyStyleModifiers(modifiers)}
`;

// for testing
Container.modifiers = modifiers;

export default Container;
