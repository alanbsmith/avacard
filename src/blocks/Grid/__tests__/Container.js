import { generateSnapshots } from "__tests__/helpers";

import Container from '../Container';

generateSnapshots(Container);
