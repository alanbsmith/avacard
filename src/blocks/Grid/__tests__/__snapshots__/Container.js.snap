// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled.div renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
}

<div
  className="c0"
/>
`;

exports[`styled.div should apply alignCenter modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`styled.div should apply flexColumn modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`styled.div should apply fluid modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem;
  padding: 0;
}

<div
  className="c0"
/>
`;
